### PR TITLE
Fw-4130, related media fields for dictionary entry and character

### PR DESCRIPTION
**Pull request by sarahfirstvoices**
### Description of Changes
- RelatedMediaMixin for models. To make this work I skipped specifying a through model or through table. Worth it?
- RelatedMediaSerializerMixin for api responses, used in character and dictionary entry APIs
- RelatedMediaTestMixin for api tests
- added prefetching with permission checking for dictionary and character APIs. (Permissions are always a good idea, but necessary in this case because media does not necessarily come from the same site.)
- bonus (?) change: split the test factories into multiple files, finally. Annoying but necessary. I commented on the few actual functional changes; the rest are just copied and pasted to different files.
- exposed a bug in the character limit validation; logged fw-4438 and disabled the relevant tests

### Checklist
- [x] README / documentation has been updated if needed
- [x] PR title / commit messages contain Jira ticket number if applicable
- [x] Tests have been updated / created if applicable
- [x] Admin Panel has been updated if models have been added or modified
- [x] Migrations have been updated and committed if applicable

### Reviewers Should Do The Following:
- [x] Review the code changes here
- [ ] Pull the branch and test locally

### Additional Notes
N/A

### CODE REVIEW

File: firstvoices/backend/tests/factories/media_factories.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
+import factory
+from factory.django import DjangoModelFactory
+
+from backend.models.media import Audio, AudioSpeaker, Image, Person, Video
+from backend.tests.factories.access import SiteFactory
+
```
Comment (sarahfirstvoices) [inline firstvoices/backend/tests/factories/media_factories.py:6]: These are the only new factories in this PR

File: firstvoices/backend/tests/factories/character_factories.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import factory
+from factory.django import DjangoModelFactory
+
+from backend.models import Alphabet, Character, CharacterVariant, IgnoredCharacter
+from backend.tests.factories import RelatedMediaBaseFactory
+from backend.tests.factories.access import SiteFactory
+
+
+class CharacterFactory(RelatedMediaBaseFactory):
```
Comment (sarahfirstvoices) [inline firstvoices/backend/tests/factories/character_factories.py:9]: Notable change to the base class for this factory

File: firstvoices/backend/tests/factories/dictionary_entry.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import factory
+
+from backend.models import DictionaryEntry
+from backend.tests.factories import RelatedMediaBaseFactory
+from backend.tests.factories.access import SiteFactory
+
+
+class DictionaryEntryFactory(RelatedMediaBaseFactory):
```
Comment (sarahfirstvoices) [inline firstvoices/backend/tests/factories/dictionary_entry.py:8]: Notable change to the base class for this factory

File: firstvoices/backend/models/media.py [hunk 4]
```diff
@@ -156,3 +156,16 @@ class Meta:
...
+
+class RelatedMediaMixin(models.Model):
+    """
+    Related media fields with standard names.
+    """
+
+    class Meta:
+        abstract = True
+
+    related_audio = models.ManyToManyField(Audio, blank=True)
```
Comment (sarahfirstvoices) [inline firstvoices/backend/models/media.py:169]: I am willing to be convinced to make this related_audios

File: firstvoices/backend/admin/characters_admin.py [hunk 5]
```diff
@@ -34,13 +34,6 @@ def formfield_for_foreignkey(self, db_field, request, **kwargs):
 
 @admin.register(Character)
 class CharacterAdmin(BaseSiteContentAdmin):
-    fields = (
-        "title",
-        "sort_order",
-        "approximate_form",
-        "site",
-        "note",
-    )
```
Comment (sarahfirstvoices) [inline firstvoices/backend/admin/characters_admin.py:43]: I removed this list so the admin shows all the model fields

**Review by NolanVH — APPROVED** [review]
These changes all look good to me. Thanks for refactoring the factories as well!
Related media fields are working as expected locally. Thanks @sarahfirstvoices.

**Pull request merged**